packages/flutter/lib/src/widgets/binding.dart

```diff
@@ -943,10 +943,39 @@ class RenderObjectToWidgetElement<T extends RenderObject> extends RootRenderObje
   }
 }
 
+/// Base class for ServiceExtensionRegistrar that provide
+/// necessary information to register a ServiceExtension.
+abstract class ExtensionRegistrarBase {
+  /// Name to registers a service extension method with.(full name
+  /// "ext.flutter.name").
+  String get name => null;
+
+  /// Callback called when the extension method is called.
+  ServiceExtensionCallback get callback => null;
+}
+
+/// Binding for service extensions to register.
+mixin ExtensionBinding on BindingBase {
+  static final List<ExtensionRegistrarBase> _extensionRegistrars = <ExtensionRegistrarBase>[];
+
+  static void registerExtensionRegistrar(ExtensionRegistrarBase registrar) {
+    _extensionRegistrars.add(registrar);
+  }
+
+  @override
+  void initServiceExtensions() {
+    super.initServiceExtensions();
+    _extensionRegistrars.forEach(_registerExtensionWithRegistrar);
```

Comment:
Why the indirection? This would probably be easier to read if you just replace it with an actual for loop that calls `registerServiceExtension(name: registrar.name, callback: registrar.callback);` directly.